pkg/cake/cake.go

```diff
@@ -0,0 +1,63 @@
+package cake
+
+type Cake struct {
+	Flour     int // grams of flour
+	Eggs      int // number of eggs
+	Butter    int // grams of butter
+	Chocolate int // grams of chocolate
+	Sugar     int // grams of sugar
+	Carrots   int // number of carrots
+	Tomatoes  int // number of tomatoes
+}
+
+// Bake takes a Cake struct and returns a string describing how this Cake tastes once baked.
+func Bake(cake Cake) string {
+	// Check for required ingredients.
+	if cake.Flour == 0 {
+		return "not a cake"
+	}
+	if cake.Eggs == 0 {
+		return "not a cake"
+	}
+	if cake.Butter == 0 {
+		return "not a cake"
+	}
+
+	if cake.Flour == 0 {
+		if cake.Eggs > 0 {
+			return "this is an omelette not a cake"
+		}
+		return "not a cake"
+	}
+
+	// If we have at least 100 grams of flour per each egg the taste is "too eggy".
+	currentFlower := cake.Flour
```

Comment:
use of "Flower" in variable name (as opposed to "Flour"), would change for consistency